Cargo.toml

```diff
@@ -33,8 +33,11 @@ rocksdb = { version = "0.14", default-features = false, features = ["snappy"], o
 cc = { version = ">=1.0.64", optional = true }
 socks = { version = "0.3", optional = true }
 lazy_static = { version = "1.4", optional = true }
+pbkdf2 = { version = "0.10", optional = true }
+unicode-normalization = { version = "0.1.19", optional = true }
+sha2 = { version = "0.10.2", optional = true }
+hmac = { version = "0.12.1", optional = true }
```

Comment:
I tried using them but `pbkfd2` wasn't accepting the rust-bitcoin `hmac` and `sha512` traits. Implementing `pbkdf2` should fix the issue.

As for porting `pbkdf2`, we could look into that. My only concern is that it may lead to bloat, especially if we simply copy it without understanding. If there are issues going forward, it will need to be maintained as well. So there's a tradeoff.